src/state.rs

```diff
@@ -109,6 +114,30 @@ pub struct ResolvedDns {
     dns_refresh_rate: std::time::Duration,
 }
 
+/// A CachedResolvedDNSMap contains do dns task for hostname via calling resolve_on_demand_dns
+#[derive(serde::Serialize, Default, Debug, Clone)]
+pub struct CachedResolvedDNSMap {
+    crd_hashmap: HashMap<String, Arc<SingleFlight>>,
+}
+
+#[derive(serde::Serialize, Default, Debug)]
+pub struct SingleFlight {
+    should_resolved_dns: AtomicBool,
+    complete_resolved_dns: AtomicBool,
```

Comment:
I donot feel comfortable with the above fields

Reply:
Hi @hzxuzhonghu, I have changed them:
1. change `should_resolved_dns` to `is_resolved_dns_running`
2. change `complete_resolved_dns` to `is_resolved_dns_completed`